app/models/procedure_revision.rb

```diff
@@ -66,22 +66,22 @@ def find_or_clone_type_de_champ(stable_id)
   end
 
   def move_type_de_champ(stable_id, position)
-    type_de_champ = find_type_de_champ_by_stable_id(stable_id)
-
-    if type_de_champ.parent.present?
-      repetition_type_de_champ = find_or_clone_type_de_champ(stable_id).parent
+    coordinate = revision_types_de_champ
+      .joins(:type_de_champ)
+      .find_by(type_de_champ: { stable_id: stable_id })
 
-      move_type_de_champ_hash(repetition_type_de_champ.types_de_champ.to_a, type_de_champ, position).each do |(id, position)|
-        type_de_champ = repetition_type_de_champ.types_de_champ.find(id)
-        type_de_champ.update!(order_place: position)
-        type_de_champ.revision_type_de_champ&.update!(position: position)
-      end
+    array = if coordinate.child?
+      revision_types_de_champ.where(parent_id: coordinate.parent_id).ordered
+    elsif coordinate.private?
+      revision_types_de_champ_private
     else
-      liste = type_de_champ.private? ? types_de_champ_private : types_de_champ_public
+      revision_types_de_champ_public
+    end.to_a
 
-      move_type_de_champ_hash(liste.to_a, type_de_champ, position).each do |(id, position)|
-        revision_types_de_champ.find_by!(type_de_champ_id: id).update!(position: position)
-      end
+    array.insert(position, array.delete_at(array.index(coordinate)))
+    array.compact.each.with_index do |e, position|
+      e.update(position: position)
+      e.type_de_champ.update!(order_place: position)
```

Comment:
```suggestion
      e.type_de_champ.update!(order_place: position) if coordinate.child?
```

et ajouter un `FIXME` que il faut faire sauter ça une fois tout est migré ?